### DISCUSSION TITLE
nlp.begin_training() causes [E050] Can't find model 'corpus/en_vectors'.

### DISCUSSION THREAD

**Discussion by WuJiunShiung** [explosion/spaCy#7093] [Help: Coding & Implementations] [answered]
Hi, guys,

I encountered an [E050] Can't find model 'corpus/en_vectors' when running nlp.begin_training(). I searched on the internet and found a discussion #3552 on a similar error. But, the error occurs even though I run nothing but nlp.begin_training(), as shown in the screenshot below.

I am using spaCy 3.0, Python 3.8 and Win 10 (64 bit).

Could someone help me with this or point me to any relevant discussion/documentation? Thank you!  

![image](https://user-images.githubusercontent.com/11987452/108066883-f1dd0380-709a-11eb-8988-28a19a71dd6b.png)

**Comment by svlandeg** [comment, marked as answer]
`begin_training` or `initialize` (it's been renamed in v3) should really only be called when you have a newly defined pipeline/component that needs initialization. In your case, you're downloading an already trained pipeline `en_core_web_md` and you shouldn't initialize it again. When you try to, it tries to find the original vectors that this pipeline was trained with, but you don't have those on your system. (we need to make this error message more clear though)

What is it that you want to do with this pipeline? If you just want to use it to make predictions, you can just call `nlp(text)` directly.

If you want to continue training on top of an already trained pipeline/component, I'd advice to have a look through the documentation [here](https://spacy.io/usage/training) first. In general we advice using the [config system](https://spacy.io/usage/training#config) for training pipelines with spaCy v3. If you want to do it in code, you'd have to call [resume_training](https://spacy.io/api/language#resume_training) instead of `begin_training` / `initialize`.

**Comment by WuJiunShiung** [comment]
I thought something was wrong with nlp.begin_training()... Anyway.
I was trying the codes of Tweet Text Classification with spaCy, a tutorial on YouTube.

There is a function as follows:

def train_spacy(train_data, iterations,test_texts,test_cats, model_arch, dropout = 0.3, model=None,init_tok2vec=None):
    ''' Train a spacy NER model, which can be queried against with test data
   
    train_data : training data in the format of (sentence, {cats: ['positive'|'negative'|'neutral']})
    labels : a list of unique annotations
    iterations : number of training iterations
    dropout : dropout proportion for training
    display_freq : number of epochs between logging losses to console
    '''
    
    nlp = spacy.load("en_core_web_md")
    

    # add the text classifier to the pipeline if it doesn't exist
    # nlp.create_pipe works for built-ins that are registered with spaCy

    if "textcat" not in nlp.pipe_names: 
        
        model = {
        "@architectures": model_arch,
        "exclusive_classes": True,
        "ngram_size": 2,
        "no_output_layer": False,
        }
        config = {"model": model}
        textcat = nlp.add_pipe("textcat", config=config)
        
               
    # otherwise, get it, so we can add labels to it
    else:
        textcat = nlp.get_pipe("textcat")

    # add label to text classifier
    textcat.add_label("positive")
    textcat.add_label("negative")
    textcat.add_label("neutral")


    # get names of other pipes to disable them during training
    pipe_exceptions = ["textcat", "trf_wordpiecer", "trf_tok2vec"]
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in pipe_exceptions]
    with nlp.disable_pipes(*other_pipes):  # only train textcat
        
        optimizer = nlp.begin_training()
        if init_tok2vec is not None:
            with init_tok2vec.open("rb") as file_:
                textcat.model.tok2vec.from_bytes(file_.read())
        print("Training the model...")
        print("{:^5}\t{:^5}\t{:^5}\t{:^5}".format("LOSS", "P", "R", "F"))
        batch_sizes = compounding(16.0, 64.0, 1.5)
        for i in range(iterations):
            print('Iteration: '+str(i))
            start_time = time.clock()
            losses = {}
            # batch up the examples using spaCy's minibatch
            random.shuffle(train_data)
            batches = minibatch(train_data, size=batch_sizes)
            for batch in batches:
                texts, annotations = zip(*batch)
                nlp.update(texts, annotations, sgd=optimizer, drop=dropout, losses=losses)
            with textcat.model.use_params(optimizer.averages):
                # evaluate on the test data 
                evaluate(nlp.tokenizer, textcat, test_texts,test_cats)
            print ('Elapsed time'+str(time.clock() - start_time)+  "seconds")
        with nlp.use_params(optimizer.averages):
            modelName = model_arch+"TweetClassification"
            filepath = "C:\\TweetSenitment\\"+modelName+"\\"
            nlp.to_disk(filepath)
    return nlp

When I run: nlp = train_spacy(training_data, 10,test_texts,test_cats,"spacy.TextCatBOW.v1"), 

I get:

----> 4 nlp = train_spacy(training_data, 10,test_texts,test_cats,"spacy.TextCatBOW.v1")
      5 

<ipython-input-12-49717ef27303> in train_spacy(train_data, iterations, test_texts, test_cats, model_arch, dropout, model, init_tok2vec)
     46     with nlp.disable_pipes(*other_pipes):  # only train textcat
     47         nlp = spacy.load("en_core_web_md") # 有個[E050] Can't find model 'corpus/en_vectors'，根據論壇，加上這行試試
---> 48         optimizer = nlp.begin_training()
     49         if init_tok2vec is not None:
     50             with init_tok2vec.open("rb") as file_:
...
... 
[E050] Can't find model 'corpus/en_vectors'. It doesn't seem to be a Python package or a valid path to a data directory.

**Reply by svlandeg** [reply]
Right, so that tutorial uses spaCy v2 and the code in it won't be directly runnable with spaCy v3. See also https://spacy.io/usage/v3

Perhaps you can ask the maker of that tutorial to publish a v3-compatible version ;-)

**Comment by WuJiunShiung** [comment]
Thank you! I will read the documentation for spaCy 3.0 and learn the new
ways.
------------------------------------------------------
Wu, Jiun-Shiung, Ph.D.
Professor and Director
Institute of Linguistics
National Chung Cheng University
Chiayi, Taiwan
Personal webpage: https://sites.google.com/view/hunterjswu
-----------------------------------------------------



Sofie Van Landeghem <notifications@github.com> 於 2021年2月18日 週四 上午5:16寫道：
